presto-hive/src/main/java/com/facebook/presto/hive/ThriftHiveMetastoreClient.java

```diff
@@ -212,6 +212,23 @@ public PrincipalPrivilegeSet getPrivilegeSet(HiveObjectRef hiveObject, String us
     }
 
     @Override
+    public void createRole(String roleName, String grantor)
+            throws TException
+    {
+        Role role = new Role();
```

Comment:
I checked hive code. Although they use the constructor you pointed, the always pass zero as a create time. I can use the same constructor actually, to be 100% compatible.